src/lib/Repository/Validator/TargetContentValidator.php

```diff
@@ -21,25 +19,21 @@
  */
 final class TargetContentValidator implements TargetContentValidatorInterface
 {
-    /** @var \eZ\Publish\API\Repository\ContentService */
-    private $contentService;
+    /** @var \eZ\Publish\API\Repository\Repository */
+    private $repository;
 
-    /** @var \eZ\Publish\API\Repository\ContentTypeService */
-    private $contentTypeService;
-
-    public function __construct(
-        ContentService $contentService,
-        ContentTypeService $contentTypeService
-    ) {
-        $this->contentService = $contentService;
-        $this->contentTypeService = $contentTypeService;
+    public function __construct(Repository $repository)
+    {
+        $this->repository = $repository;
     }
 
     public function validate(int $value, array $allowedContentTypes = []): ?ValidationError
     {
         try {
-            $contentInfo = $this->contentService->loadContentInfo($value);
-            $contentType = $this->contentTypeService->loadContentType($contentInfo->contentTypeId);
+            $contentInfo = $this->repository->sudo(static function (Repository $repository) use ($value) {
+                return $repository->getContentService()->loadContentInfo($value);
+            });
+            $contentType = $this->repository->getContentTypeService()->loadContentType($contentInfo->contentTypeId);
```

Comment:
1) return type is missing in callback 
2) Please consider using \eZ\Publish\SPI\Persistence\Content\Handler::loadContentInfo and \eZ\Publish\SPI\Persistence\Content\Type\Handler::load instead of `sudo`